.github/workflows/test.yaml

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 DigitalOcean
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### Run unit as well as end-to-end tests on PRs and master pushes. For the
+### latter, we also push a new CSI plugin master image to the container
+### registry.
+name: test
+
+env:
+  DOCKER_ORG: digitalocean
+  DOCKER_USER: ${{ secrets.DockerHubUser }}
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+
+      - name: Go setup
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.13.5'
+
+      - name: Run unit tests
+        run: make
+
+  push-images:
+    runs-on: ubuntu-18.04
+    needs: unit-test
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+
+      - name: Log into container registry
+        run: echo ${{ secrets.DockerHubToken }} | docker login --username ${DOCKER_USER} --password-stdin
+
+      - name: build and push runner image
```

Comment:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths

Maybe we can limit when this actually triggers with the `path` option? Although I see you've done some smart `--from-cache` work with the build so it may not be needed

Reply:
Unfortunately, the path-based trigger does not work at the job or step level, and there's no way to describe dependencies between workflows, so we cannot use it.